packages/flutter/lib/src/material/input_decorator.dart

```diff
@@ -1574,7 +1558,7 @@ class _RenderDecoration extends RenderBox with SlottedContainerRenderObjectMixin
       final double t = decoration.floatingLabelProgress;
       // The center of the outline border label ends up a little below the
       // center of the top border line.
-      final bool isOutlineBorder = decoration.border != null && decoration.border.isOutline;
+      final bool isOutlineBorder = decoration.border.isOutline;
```

Comment:
Note to self to go back and check decoration.border is now non-nullable.

Reply:
And it is:
https://github.com/flutter/flutter/pull/119022/files/c071acf07b6d8b18000796c60d1b8d3b749d447d#diff-478235b366f9f8e527d852fd87629c538ed2123c6c80c21113a4d943b2ab15cfR599